src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,97 @@
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        ///   Exposes methods to set goal time <see cref="Goal"/>
+        /// </summary>
+        /// <param name="ChannelName"></param>
+        /// <param name="DefinitionId"></param>
+        /// <param name="Minutes"></param>
+        [HttpPost]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.Created, Type = typeof(Models.Goal), Description = "Goal for given Channel and Definition-Id is created/updated")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([Required] int DefinitionId, [Required] string ChannelName, [Required] int Minutes)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                    .Where(c => c.Name.Equals(ChannelName)).FirstOrDefaultAsync();
+            Data.Models.GoalTime goal = await _context.GoalTime
+                .Where(g => g.DefinitionId == DefinitionId && g.ChannelId == channel.Id).FirstOrDefaultAsync();
+
+            if (channel == null)
+            {
+                return NotFound();
+            }
+            if (goal == null)
+            {
+                var goalModel = new Data.Models.GoalTime
+                {
+                    DefinitionId = DefinitionId,
+                    Minutes = Minutes,
+                    ChannelId = channel.Id
+                };
+                await _context.GoalTime.AddAsync(goalModel);
+                await _context.SaveChangesAsync();
+                var goalTest = CreatedAtRoute(
+                    new
+                    {
+                        action = "CreateGoal",
+                        DefinitionId = DefinitionId,
+                        Minutes = Minutes,
+                        ChannelId = channel.Id
+                    },
+                    new Goal(goalModel));
+                return StatusCode((int)HttpStatusCode.Created);
+            }
+            else
+            {
+                goal.Minutes = Minutes;
+                _context.GoalTime.Update(goal);
+                await _context.SaveChangesAsync();
+
+                return Ok(new Goal(goal));
+            }
+        }
+
+        /// <summary>
+        /// Get goal time in minutes <see cref="Goal"/>s that match the given classification.
+        /// </summary>
+        /// <param name="ChannelName"></param>
+        /// <param name="DefinitionId"></param>
+        [HttpGet]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(Goal), Description = "Get the Goal for given Channel and Definition-Id")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> GetGoalTimes([Required]int DefinitionId, [Required]string ChannelName)
+        {
+            Data.Models.Channel channel = await _context.Channels
```

Comment:
The whitespace in this method is uneven.

Reply:
I removed all the trailing spaces

Reply:
The leading whitespace is what's weird. Line 79 is double indented when it should be one. Line 85 is not indented at all, when it should also be one.

Reply:
got it. For some reason no matter how many times I format the file. It does not work. Like in Eclipse the way to intend is so damn different than in visual studio. Sorry about this, I need to fig out a better way for correcting this indentation in VS.

Reply:
VS without extensions doesn't actually care about indentation of continued lines. Our guideline is indent every continued line one indent more than the line that is being continued.